serde_derive/src/ser.rs

```diff
@@ -599,11 +614,12 @@ fn serialize_internally_tagged_variant(
 
     match effective_style(variant) {
         Style::Unit => {
+            let variant_name = serialize_variant_name(&variant_name);
             quote_block! {
                 let mut __struct = _serde::Serializer::serialize_struct(
                     __serializer, #type_name, 1)?;
                 _serde::ser::SerializeStruct::serialize_field(
-                    &mut __struct, #tag, #variant_name)?;
+                    &mut __struct, #tag, &#variant_name)?;
```

Comment:
This should not need to go through `_serde::__private::ser::VariantName`. You can directly serialize `&true` or `&1` as the value here.

Same comment may apply elsewhere in this file. Please use `VariantName` only where needed for `serialize_tagged_newtype` calls.